modules/ConnectedRouter.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component, PropTypes } from 'react'
+import BrowserHistory from 'react-history/BrowserHistory'
+import StaticRouter from 'react-router/StaticRouter'
+
+import { LOCATION_CHANGE } from './reducer'
+
+class ConnectedRouter extends Component {
+  static propTypes = {
+    store: PropTypes.object,
+    history: PropTypes.func,
+    basename: PropTypes.string,
+    keyLength: PropTypes.number
+  }
+
+  static contextTypes = {
+    store: PropTypes.object
+  }
+
+  static defaultProps = {
+    history: BrowserHistory
+  }
+
+  render() {
+    const { history:History, basename, keyLength, ...props } = this.props
+
+    return (
+      <History location={location} basename={basename} keyLength={keyLength}>
+        {({ history, action, location }) => {
+          const store = this.context.store || this.props.store
+
+          store.dispatch({
+            type: LOCATION_CHANGE,
+            payload: { action, location }
+          })
```

Comment:
Won't this cause infinite re-renders? It's firing even if the location is the same and that will cause another render.